zkevm-circuits/src/witness/mpt.rs

```diff
@@ -201,6 +250,40 @@ impl MptUpdates {
         Self::from_rws_with_mock_state_roots(rows, 0xcafeu64.into(), 0xdeadbeefu64.into())
     }
 
+    pub(crate) fn from_unsorted_rws_with_mock_state_roots(
+        rows: &[Rw],
+        old_root: U256,
+        new_root: U256,
+    ) -> Self {
+        log::debug!("mpt update roots (mocking) {:?} {:?}", old_root, new_root);
+        let rows_len = rows.len();
+        let mut updates = BTreeMap::new(); // TODO: preallocate
+        for (key, row) in rows.iter().filter_map(|row| key(row).map(|key| (key, row))) {
+            updates.entry(key).or_insert_with(Vec::new).push(*row); // TODO: preallocate
+        }
+        let updates: BTreeMap<_, _> = updates
+            .into_iter()
+            .enumerate()
+            .map(|(i, (key, rows))| {
+                MptUpdate::from_rows(key, rows, i, rows_len, old_root, new_root)
+            })
+            .collect();
+        let mpt_updates = MptUpdates {
+            updates,
+            old_root,
+            new_root,
+            ..Default::default()
+        };
+        #[cfg(debug_assertions)]
```

Comment:
add a comment like // FIXME: we can remove this assert after the code runs a while and everthing is ok?